src/my-server.js

```diff
@@ -36,7 +35,7 @@ class Server {
         }
         catch (err) {
             error(`Server failed to start`);
-            throw err;
+            error(err);
```

Comment:
Printing an original error is a wise decision, but looks like we forget to pass an error to the caller.

Reply:
But this problem refer to `Server`, `App` only uses it interface, so shouldn't we encapsulate it too inside our `Server`?

Reply:
A problem in your server means a problem with your app. A problem in your app is your users' problem. It's better when users are aware of problems, so they can take corresponding actions: filing a bug report, searching for workarounds, debugging the app.